scripts/sync_prod

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+#
+# sync_prod - 0.1
+#
+#
+# sync_prod is a utility to sync database dumps from production and and to
+# sync media files (if applicable) for a Django application that are
+# user managed (i.e. uploaded photos for a CMS). Assumes that a proper
+# .my.cnf with credentials exists on target hosts for mysqldump.
+#
+# Dependencies:
+# - jq (system install via package manager for JSON parsing)
+#
+# Usage: sync_prod [-h] [-d] config.json
+# -h    Display this help message
+# -d    Dry-run
+
+printUsage() {
+  echo "Usage: ${0} [-h] [-d] config.json"
+  echo "-h    Display this help message"
+  echo "-d    Dry-run"
+}
+
+# Parse usage option flags, with \? as a backstop against invalid options.
+while getopts ":hd" opt; do
+  case ${opt} in
+    h )
+      printUsage
+      exit 0
+      ;;
+    d )
+      dryrun=true
+      ;;
+    \? )
+      echo "Invalid option: -${OPTARG}" 1>&2
+      printUsage 1>&2
+      exit 1
+      ;;
+  esac
+done;
+# Shift past any options that were consumed
+shift $((OPTIND -1))
+
+# Check if there is still a path to config.json
+if [ -z "${1}" ]
+    then
+      echo "No config file supplied."
+      echo ""
+      printUsage 1>&2
+      exit 1
+fi
+
+# Configurations
+config="$1"
+# Use tmp for temporary dumps
+tmp_file=/tmp/db_tmp.sql
+applications=()
+# Get configured storage dir, or if not, assume PWD
+storageDir=$(jq -r ".storageDir" "${config}")
+if [[ storageDir == "null" ]]
+  then
+    storageDir=${PWD}
+fi
+
+# create a storage dir just in case it doesn't exist
+mkdir -p "${storageDir}"
+
+# This loop avoids issues with spaces in values, although they probably
```

Comment:
It's a tiny idiosyncrasy of bash. I can't come up with a reason why, but if we had spaces in values, it would break the way bash parses array values using `()` via a space delimited list. By passing in a loop, that doesn't happen.